server/core/src/main/java/dev/slimevr/tracking/processor/skeleton/HumanSkeleton.kt

```diff
@@ -1736,11 +1736,48 @@ class HumanSkeleton(
 	}
 
 	companion object {
+		// I don't know why this works but let's just leave it for now lol, see UP_ADJ
+		// - Butterscotch
 		val FORWARD_QUATERNION = EulerAngles(
 			EulerOrder.YZX,
 			FastMath.HALF_PI,
 			0f,
 			0f,
 		).toQuaternion()
+
+		/**
+		 * Used to rotate an identity quaternion to face up.
+		 */
+		val UP_ADJ = EulerAngles(
+			EulerOrder.YZX,
+			-FastMath.HALF_PI,
+			0f,
+			0f,
+		).toQuaternion()
+
+		/**
+		 * Similar to twinNearest, but uses the longest rotation for the lower back
+		 * quadrant. This handles the thigh extending behind the torso to face
+		 * downwards, and the hip extending behind the chest. The thigh cannot bend to
+		 * the back away from the torso and the spine hopefully can't bend back that far,
+		 * so we can fairly safely assume the rotation is towards the torso.
+		 */
+		fun signExtendedRot(ref: Quaternion, extended: Quaternion?): Quaternion {
+			// Handle null here for convenience
+			if (extended == null) {
+				return ref
+			}
+
+			val isBack = ref.angleToR(extended) > FastMath.HALF_PI
+			val isLower = (ref * UP_ADJ).angleToR(extended) > FastMath.HALF_PI
+
+			return if (isBack && isLower) {
+				// Select longest rotation
+				if (extended.dot(ref) >= 0f) -extended else extended
+			} else {
+				// Select shortest rotation
+				extended.twinNearest(ref)
+			}
+		}
```

Comment:
maybe we should move this to `Quaternion`?

Reply:
Hmm perhaps... This is very skeleton specific, but also that would make more logical sense, I think it's a good idea

Reply:
I think this is good yeah? I also realized I can just use `twinNearest` directly lol this code is so much simpler